extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -291,6 +294,69 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 		return this._assessmentResults;
 	}
 
+	public async getSkuRecommendations(
+		dataFolder: string,
+		perfQueryIntervalInSec: number,
+		targetPlatforms: MigrationTargetType[],
+		targetPercentile: number,
+		scalingFactor: number,
+		startTime: string,
+		endTime: string,
+		databaseAllowList: string[]): Promise<SkuRecommendation> {
+		try {
+			const serverInfo = await azdata.connection.getServerInfo(this.sourceConnectionId);
+			const machineName = (<any>serverInfo)['machineName'];		// get actual machine name instead of whatever the user entered as the server name (e.g. DESKTOP-xxx instead of localhost)
```

Comment:
I think you should be able to get the server name with `serverInfo.serverName`
reference: https://github.com/microsoft/azuredatastudio/blob/b39025ddbb400a8b01c637ea9863c1ee4a4ef0c9/extensions/sql-migration/src/models/stateMachine.ts#L933

Reply:
I tried that originally, but that one always points to whatever server name was given when the connection was first added. So for example if I decided to connect to my local instance using 'localhost' as the server name then `serverInfo.serverName` would also be 'localhost', but for SKU recommendation we need the full machine/instance name (e.g. 'DESKTOP-ratruong' or whatever) so I had to use this one instead